src/request.rs

```diff
@@ -506,6 +511,30 @@ impl<State> Request<State> {
             .and_then(|cookie_data| cookie_data.content.read().unwrap().get(name).cloned())
     }
 
+    #[cfg(feature = "sessions")]
+    pub fn session(&self) -> &crate::sessions::Session {
+        self.ext::<crate::sessions::Session>().expect(
+            "request session not initialized, did you enable tide::sessions::SessionMiddleware?",
+        )
+    }
+
+    #[cfg(feature = "sessions")]
+    pub fn session_mut(&mut self) -> &mut crate::sessions::Session {
+        self.ext_mut().expect(
+            "request session not initialized, did you enable tide::sessions::SessionMiddleware?",
+        )
+    }
+
+    #[cfg(feature = "sessions")]
+    pub fn session_get<T: serde::de::DeserializeOwned>(&self, key: &str) -> Option<T> {
+        self.session().get(key)
+    }
+
+    #[cfg(feature = "sessions")]
+    pub fn session_insert(&mut self, key: &str, value: impl serde::Serialize) -> crate::Result<()> {
+        Ok(self.session_mut().insert(key, value)?)
+    }
```

Comment:
I'm wondering whether we should expose these methods. These methods are also directly available on the session object. Perhaps we could omit these for now and only add them back if we see they're frequently used? That way we don't add a lot of methods to `Request` all at once.

Reply:
Totally agree. It feels a bit like a law of Demeter thing, but in theory ext has the same problem — we don't do .ext().get() and .ext_mut().insert(…), we do .ext() and .set_ext(…). Wasn't sure if I should stick with that convention or encourage people to chain off of session() and session_mut(). Relatedly, the mutability in insert and remove isn't strictly necessary because it's all interior mutation. I just added the mutability constraint so future changes that eliminate the interior mutability don't require a breaking change.

Reply:
Removed it for now, can always add it back in on a minor release